src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -754,7 +754,11 @@ private void handleEditorTweaksReceived(JSONObject tweaksMessage) {
         private void handleVariantsReceived(JSONArray variants) {
             final SharedPreferences preferences = getSharedPreferences();
             final SharedPreferences.Editor editor = preferences.edit();
-            editor.putString(SHARED_PREF_CHANGES_KEY, variants.toString());
+            if(variants != null) {
```

Comment:
I am not a fan of using `null` as a meaningful value that will change the behavior/flow of the product. For instance, `variants` could be just an empty `JSONArray` if there are really no `variants`. The `SHARED_PREF_CHANGES_KEY` could always be in the device, just with empty variants.